python/ray/train/tests/test_backend.py

```diff
@@ -106,6 +123,71 @@ def on_shutdown(self, worker_group: WorkerGroup, backend_config: TestConfig):
         pass
 
 
+class MockWorkerGroup(WorkerGroup):
+    # Mock construct metadata
+    def add_workers(self, num_workers: int):
```

Comment:
I don't think we need to override `add_workers` right? Just need to override `construct_metadata`

Reply:
`construct_metadata` is not a class function and is called inside the `add_workers` so I just hardcoded inside the `add_workers`

Reply:
See the comment below, but we should be able to patch the `construct_metadata` function with our own mock.

Reply:
Done